### ISSUE TITLE
Builds on Vercel are broken

### ISSUE THREAD

**Issue by borekb** [squidfunk/mkdocs-material#1690]
I just want to report a strange behavior I'm seeing with some deployments to Vercel. It might be a Vercel issue, it might be a MkDocs issue or it might be an issue of this theme, I'm really not sure so please bear with me.

I'm creating a docs site that's in its early stages here: https://docs-aouokdqlf.now.sh/. I cannot share the repo but I've created a clone with basically an identical setup:

https://github.com/borekb/mkdocs-vercel-demo

This **demo repo** always works for me but with our **real** docs site, when the Vercel service does the build (i.e., Vercel runs `mkdocs build -d output`), the deployments are broken. In fact, so broken that my Chrome becomes basically unresponsive and Firefox wants to kill the site after a few moments:

<img width="1438" alt="Screenshot 2020-05-13 at 10 36 02" src="https://user-images.githubusercontent.com/101152/81790359-9414fd00-9505-11ea-8475-4dd26bf760af.png">

An example of such broken deployment is https://docs-45uy3tha0.now.sh/. Note how the UI is rendered but even hovering over links doesn't make them clickable. Something is very broken there and at first debugging attempts, I didn't figure out what.

A couple of observations:

- Since the demo repo deploys fine, can it be that our contents is causing this? Maybe the built-in search indexes some strange unicode character (emoji) and then crashes? But perhaps unlikely because:
- In our real repo, I _can_ do this:

```bash
# Build to `output`
docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material build -d public

# Start a local dev server and browse the output
cd public
ws   # This is https://www.npmjs.com/package/local-web-server
→ Docs running at http://localhost:8000
```

These docs run perfectly fine.

- Can it be that my `requirements.txt` are different than what is in Dockerfile and that's the cause of the Vercel build failing? (I copied [`requirements.txt` from the current master](https://github.com/squidfunk/mkdocs-material/blob/b8cf7d76e5c9cc5e97cb21ebb5342b2265b510c2/requirements.txt) in this repo though.)

### Project configuration

<details><summary>Our <code>mkdocs.yml</code></summary>

```yaml
site_name: SOFA docs
docs_dir: content

# Keep this in sync with `content/index.md` (tip: copy/paste from there & apply some VSCode-fu)
nav:
  - 'Home': index.md
  - 'Introduction':
      - 'What is Shoptet Premium': introduction/shoptet-premium.md
      - 'What is SOFA': introduction/sofa.md
  - 'Tutorial':
      - '01: Create project': tutorial/01-create-project.md
      - '02: Build UI': tutorial/02-build-ui.md
      - '03: Fetch data': tutorial/03-fetch-data.md
      - '04: Implement cart': tutorial/04-implement-cart.md
      - '05: Implement checkout': tutorial/05-implement-checkout.md
      - '06: Deploy to lab': tutorial/06-deploy-to-lab.md
      - '07: Deploy to prod': tutorial/07-deploy-to-prod.md
  - 'Guides':
      - 'TypeScript': guides/typescript.md
      - 'Routing': guides/routing.md
      - 'Data fetching': guides/data-fetching.md
      - 'Styling': guides/styling.md
      - 'Images': guides/images.md
      - 'Localization': guides/localization.md
      - 'Structured data': guides/structured-data.md
      - 'GTM + data layer': guides/gtm-data-layer.md
      - 'Cart & checkout': guides/cart-checkout.md
      - 'Tooling': guides/tooling.md
      - 'Advanced GraphQL topics': guides/advanced-graphql.md
      - 'Deployments': guides/deployments.md

theme:
  name: material
  custom_dir: mkdocs-overrides
  features:
    - instant
  palette:
    primary: white
    accent: indigo
  logo: https://user-images.githubusercontent.com/101152/81295258-701e6b00-9070-11ea-9fbc-3f7d0c07c6d3.png

extra_css:
  - mkdocs.css

markdown_extensions:
  - admonition
  - codehilite:
      guess_lang: false
  - toc:
      permalink: true
  - pymdownx.tasklist:
      custom_checkbox: true
```

</details>

### Package versions

Locally, I use Docker. When sent to Vercel, this `requirements.txt` is used:

https://github.com/borekb/mkdocs-vercel-demo/blob/7b5f816a9689a633e3bcd07d9f4f71bab53f3822/requirements.txt

**Comment by borekb** [comment]
I'll also post build logs for that problematic deployment https://docs-45uy3tha0.now.sh/ – everything looks fine, from MkDocs installation to the final `mkdocs build` command.

```
09:50:31.905  Retrieving list of deployment files...
09:50:32.080  Downloading 87 deployment files...
09:50:38.270  Installing build runtime...
09:50:38.724  Build runtime installed: 453.664ms
09:50:39.048  Looking up build cache...
09:50:39.160  Build cache found. Downloading...
09:50:39.168  Build cache downloaded [220.00 B]: 7.728ms
09:50:39.923  Collecting mkdocs>=1.1
09:50:39.980    Downloading mkdocs-1.1-py2.py3-none-any.whl (6.4 MB)
09:50:40.264  Collecting Pygments>=2.4
09:50:40.275    Downloading Pygments-2.6.1-py3-none-any.whl (914 kB)
09:50:40.373  Collecting markdown>=3.2
09:50:40.384    Downloading Markdown-3.2.2-py3-none-any.whl (88 kB)
09:50:40.445  Collecting pymdown-extensions>=7.0
09:50:40.458    Downloading pymdown_extensions-7.1-py2.py3-none-any.whl (205 kB)
09:50:40.497  Collecting mkdocs-material-extensions>=1.0b1
09:50:40.509    Downloading mkdocs_material_extensions-1.0b2-py3-none-any.whl (8.0 kB)
09:50:40.550  Collecting livereload>=2.5.1
09:50:40.561    Downloading livereload-2.6.1-py2.py3-none-any.whl (23 kB)
09:50:40.610  Collecting click>=3.3
09:50:40.621    Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
09:50:40.680  Collecting tornado>=5.0
09:50:40.693    Downloading tornado-6.0.4.tar.gz (496 kB)
09:50:41.036  Collecting PyYAML>=3.10
09:50:41.047    Downloading PyYAML-5.3.1.tar.gz (269 kB)
09:50:41.370  Collecting lunr[languages]==0.5.6
09:50:41.382    Downloading lunr-0.5.6-py2.py3-none-any.whl (36 kB)
09:50:41.431  Collecting Jinja2>=2.10.1
09:50:41.441    Downloading Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
09:50:41.515  Collecting importlib-metadata; python_version < "3.8"
09:50:41.525    Downloading importlib_metadata-1.6.0-py2.py3-none-any.whl (30 kB)
09:50:41.622  Collecting mkdocs-material>=5.0.0
09:50:41.644    Downloading mkdocs_material-5.1.6-py2.py3-none-any.whl (3.6 MB)
09:50:42.535  Collecting six
09:50:42.546    Downloading six-1.14.0-py2.py3-none-any.whl (10 kB)
09:50:42.580  Collecting future>=0.16.0
09:50:42.592    Downloading future-0.18.2.tar.gz (829 kB)
09:50:42.962  Collecting nltk>=3.2.5; extra == "languages"
09:50:42.976    Downloading nltk-3.5.zip (1.4 MB)
09:50:43.338  Collecting MarkupSafe>=0.23
09:50:43.349    Downloading MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl (27 kB)
09:50:43.391  Collecting zipp>=0.5
09:50:43.402    Downloading zipp-3.1.0-py3-none-any.whl (4.9 kB)
09:50:43.470  Collecting joblib
09:50:43.482    Downloading joblib-0.14.1-py2.py3-none-any.whl (294 kB)
09:50:44.002  Collecting regex
09:50:44.014    Downloading regex-2020.5.7-cp36-cp36m-manylinux2010_x86_64.whl (675 kB)
09:50:44.149  Collecting tqdm
09:50:44.160    Downloading tqdm-4.46.0-py2.py3-none-any.whl (63 kB)
09:50:44.288  Installing collected packages: tornado, six, livereload, click, zipp, importlib-metadata, markdown, PyYAML, future, joblib, regex, tqdm, nltk, lunr, MarkupSafe, Jinja2, mkdocs, Pygments, pymdown-extensions, mkdocs-material, mkdocs-material-extensions
09:50:44.288      Running setup.py install for tornado: started
09:50:44.805      Running setup.py install for tornado: finished with status 'done'
09:50:44.949      Running setup.py install for PyYAML: started
09:50:45.236      Running setup.py install for PyYAML: finished with status 'done'
09:50:45.239      Running setup.py install for future: started
09:50:45.753      Running setup.py install for future: finished with status 'done'
09:50:46.036      Running setup.py install for nltk: started
09:50:46.941      Running setup.py install for nltk: finished with status 'done'
09:50:49.768  Successfully installed Jinja2-2.11.2 MarkupSafe-1.1.1 PyYAML-5.3.1 Pygments-2.6.1 click-7.1.2 future-0.18.2 importlib-metadata-1.6.0 joblib-0.14.1 livereload-2.6.1 lunr-0.5.6 markdown-3.2.2 mkdocs-1.1 mkdocs-material-5.1.6 mkdocs-material-extensions-1.0b2 nltk-3.5 pymdown-extensions-7.1 regex-2020.5.7 six-1.14.0 tornado-6.0.4 tqdm-4.46.0 zipp-3.1.0
09:50:49.972  Installing dependencies...
09:50:50.177  yarn install v1.22.0
09:50:50.195  info No lockfile found.
09:50:50.200  [1/4] Resolving packages...
09:50:50.201  [2/4] Fetching packages...
09:50:50.203  [3/4] Linking dependencies...
09:50:50.206  [4/4] Building fresh packages...
09:50:50.208  success Saved lockfile.
09:50:50.212  Done in 0.04s.
09:50:50.219  Running "yarn run build"
09:50:50.379  yarn run v1.22.0
09:50:50.399  $ mkdocs build -d public
09:50:50.799  INFO    -  Cleaning site directory 
09:50:50.801  INFO    -  Building documentation to directory: /zeit/56d9bcad/public 
09:50:51.181  INFO    -  Documentation built in 0.38 seconds 
09:50:51.220  Done in 0.84s.
09:50:52.085  Uploading build outputs...
09:50:56.713  Build completed. Populating build cache...
09:50:56.787  Uploading build cache [220.00 B]...
09:50:56.890  Build cache uploaded: 103.391ms
09:50:56.896  Done with "package.json"
```

It's also worth noting that I've seen successful deployments using this method, for example, this one works: https://docs-7j52vzwfc.now.sh/ and its build logs are:

```
20:30:02.294  Retrieving list of deployment files...
20:30:02.455  Downloading 25 deployment files...
20:30:05.829  Installing build runtime...
20:30:06.238  Build runtime installed: 408.652ms
20:30:06.574  Looking up build cache...
20:30:06.638  Build cache found. Downloading...
20:30:06.646  Build cache downloaded [206.00 B]: 7.522ms
20:30:07.335  Collecting mkdocs>=1.1
20:30:07.391    Downloading mkdocs-1.1-py2.py3-none-any.whl (6.4 MB)
20:30:07.901  Collecting Pygments>=2.4
20:30:07.917    Downloading Pygments-2.6.1-py3-none-any.whl (914 kB)
20:30:08.029  Collecting markdown>=3.2
20:30:08.041    Downloading Markdown-3.2.1-py2.py3-none-any.whl (88 kB)
20:30:08.098  Collecting pymdown-extensions>=7.0
20:30:08.113    Downloading pymdown_extensions-7.1-py2.py3-none-any.whl (205 kB)
20:30:08.153  Collecting mkdocs-material-extensions>=1.0b1
20:30:08.165    Downloading mkdocs_material_extensions-1.0b2-py3-none-any.whl (8.0 kB)
20:30:08.206  Collecting lunr[languages]==0.5.6
20:30:08.219    Downloading lunr-0.5.6-py2.py3-none-any.whl (36 kB)
20:30:08.278  Collecting tornado>=5.0
20:30:08.293    Downloading tornado-6.0.4.tar.gz (496 kB)
20:30:08.632  Collecting Jinja2>=2.10.1
20:30:08.644    Downloading Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
20:30:08.700  Collecting click>=3.3
20:30:08.713    Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
20:30:08.748  Collecting livereload>=2.5.1
20:30:08.763    Downloading livereload-2.6.1-py2.py3-none-any.whl (23 kB)
20:30:08.821  Collecting PyYAML>=3.10
20:30:08.835    Downloading PyYAML-5.3.1.tar.gz (269 kB)
20:30:09.135  Requirement already satisfied: setuptools>=36 in /usr/local/lib/python3.6/site-packages (from markdown>=3.2->-r /zeit/1b50d185/requirements.txt (line 5)) (40.6.2)
20:30:09.220  Collecting mkdocs-material>=5.0.0
20:30:09.240    Downloading mkdocs_material-5.1.5-py2.py3-none-any.whl (3.6 MB)
20:30:10.189  Collecting future>=0.16.0
20:30:10.202    Downloading future-0.18.2.tar.gz (829 kB)
20:30:10.574  Collecting six>=1.11.0
20:30:10.587    Downloading six-1.14.0-py2.py3-none-any.whl (10 kB)
20:30:10.625  Collecting nltk>=3.2.5; extra == "languages"
20:30:10.642    Downloading nltk-3.5.zip (1.4 MB)
20:30:11.020  Collecting MarkupSafe>=0.23
20:30:11.034    Downloading MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl (27 kB)
20:30:11.100  Collecting joblib
20:30:11.117    Downloading joblib-0.14.1-py2.py3-none-any.whl (294 kB)
20:30:11.646  Collecting regex
20:30:11.667    Downloading regex-2020.5.7-cp36-cp36m-manylinux2010_x86_64.whl (675 kB)
20:30:11.809  Collecting tqdm
20:30:11.822    Downloading tqdm-4.46.0-py2.py3-none-any.whl (63 kB)
20:30:11.938  Installing collected packages: future, six, click, joblib, regex, tqdm, nltk, lunr, tornado, MarkupSafe, Jinja2, markdown, livereload, PyYAML, mkdocs, Pygments, pymdown-extensions, mkdocs-material, mkdocs-material-extensions
20:30:11.939      Running setup.py install for future: started
20:30:12.449      Running setup.py install for future: finished with status 'done'
20:30:12.793      Running setup.py install for nltk: started
20:30:13.731      Running setup.py install for nltk: finished with status 'done'
20:30:13.778      Running setup.py install for tornado: started
20:30:14.302      Running setup.py install for tornado: finished with status 'done'
20:30:14.454      Running setup.py install for PyYAML: started
20:30:14.753      Running setup.py install for PyYAML: finished with status 'done'
20:30:17.516  Successfully installed Jinja2-2.11.2 MarkupSafe-1.1.1 PyYAML-5.3.1 Pygments-2.6.1 click-7.1.2 future-0.18.2 joblib-0.14.1 livereload-2.6.1 lunr-0.5.6 markdown-3.2.1 mkdocs-1.1 mkdocs-material-5.1.5 mkdocs-material-extensions-1.0b2 nltk-3.5 pymdown-extensions-7.1 regex-2020.5.7 six-1.14.0 tornado-6.0.4 tqdm-4.46.0
20:30:17.720  Installing dependencies...
20:30:17.930  yarn install v1.22.0
20:30:17.948  info No lockfile found.
20:30:17.953  [1/4] Resolving packages...
20:30:17.954  [2/4] Fetching packages...
20:30:17.956  [3/4] Linking dependencies...
20:30:17.959  [4/4] Building fresh packages...
20:30:17.962  success Saved lockfile.
20:30:17.965  Done in 0.04s.
20:30:17.972  Running "yarn run build"
20:30:18.132  yarn run v1.22.0
20:30:18.153  $ mkdocs build -d public
20:30:18.540  INFO    -  Cleaning site directory 
20:30:18.540  INFO    -  Building documentation to directory: /zeit/1b50d185/public 
20:30:18.835  INFO    -  Documentation built in 0.30 seconds 
20:30:18.870  Done in 0.74s.
20:30:19.730  Uploading build outputs...
20:30:25.030  Build completed. Populating build cache...
20:30:25.102  Uploading build cache [220.00 B]...
20:30:25.156  Build cache uploaded: 53.558ms
20:30:25.161  Done with "package.json"
```

When I delete the timestamps and compare the build logs (working on the left, broken on the right), it looks quite similar but maybe the problem is somewhere in there?

![Screenshot 2020-05-13 at 11 06 37](https://user-images.githubusercontent.com/101152/81793628-eeb05800-9509-11ea-89c8-1e271feead7e.png)

**Comment by borekb** [comment]
I've also opened a Vercel discussion: https://github.com/zeit/now/discussions/4364.

**Comment by squidfunk** [comment]
I don't know what Vercel is and never used it, but as he [official docs](https://squidfunk.github.io/mkdocs-material/) work perfectly fine on GitHub pages, I'd say it's a downstream issue. 

Also, you clearly have made some customizations and/or use some plugins/scripts. Does this happen with a mint installation?

**Comment by borekb** [comment]
Ok so it turns out that the problem happens only with [instant loading](https://squidfunk.github.io/mkdocs-material/getting-started/#instant-loading), which I happen to have turned on:

```yaml
theme:
  features:
    - instant
```

When I remove it, everything is fine.

Also fine is when I build using the official Docker container.

Only when the dependencies are installed manually on Vercel from `requirements.txt`, which currently contains this:

https://github.com/squidfunk/mkdocs-material/blob/b8cf7d76e5c9cc5e97cb21ebb5342b2265b510c2/requirements.txt#L22-L26

the resulting site is broken.

I'll try to install locally to see if I can replicate via local, non-Dockerized build.

**Comment by borekb** [comment]
Well, I'm not sure how to work with Python correctly, I tried `pipenv install mkdocs` as indicated in Getting Started but got this:

```
✔ Successfully created virtual environment!
Virtualenv location: /Users/borekb/.local/share/virtualenvs/docs-AogWyTTU
Creating a Pipfile for this project…
Installing mkdocs-material…
Adding mkdocs-material to Pipfile's [packages]…
✔ Installation Succeeded
Pipfile.lock not found, creating…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
✘ Locking Failed!
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/resolver.py", line 61, in resolve
[pipenv.exceptions.ResolutionFailure]:       return resolve_deps(
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/utils.py", line 718, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree, hashes, markers_lookup, resolver = actually_resolve_deps(
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches mkdocs-material-extensions>=1.0b1
[pipenv.exceptions.ResolutionFailure]:       Skipped pre-versions: 1.0b1, 1.0b1, 1.0b2, 1.0b2
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches mkdocs-material-extensions>=1.0b1
Skipped pre-versions: 1.0b1, 1.0b1, 1.0b2, 1.0b2
There are incompatible versions in the resolved dependencies.
[pipenv.exceptions.ResolutionFailure]:       return resolve_deps(
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/utils.py", line 718, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree, hashes, markers_lookup, resolver = actually_resolve_deps(
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2018.11.26_4/libexec/lib/python3.8/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches mkdocs-material-extensions>=1.0b1
[pipenv.exceptions.ResolutionFailure]:       Skipped pre-versions: 1.0b1, 1.0b1, 1.0b2, 1.0b2
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches mkdocs-material-extensions>=1.0b1
Skipped pre-versions: 1.0b1, 1.0b1, 1.0b2, 1.0b2
There are incompatible versions in the resolved dependencies.
```

I have to admin that I'm a complete Python noob (that's why I use Docker image instead, locally).

---

Also, as someone coming from the JS/npm ecosystem, I'm trying to find a file in this repo which would lock specific versions of dependencies. I see `requirements.txt` which is an equivalent of `package.json` I guess but I don't see any equivalent of `package-lock.json` / `yarn.lock`.

When I push the project to Vercel, and it only contains `requirements.txt`, different sets of dependencies might get installed for different deployments, which is generally relatively unsafe. Can I possibly specify exact versions in `requirements.txt`? How do I find out what those should be?

**Comment by squidfunk** [comment]
If it's related to instant loading, it won't matter which version of other packages are installed. However, we need a locally reproducible case to help you. Also, note that instant loading is still considered experimental.

Could you try and deploy your site on, say, GitHub pages? Or does it only happen with Vercel?

**Comment by borekb** [comment]
I'd love to try building it locally but don't know how to install it. `pipenv install mkdocs-material` should be enough, right?

I'm sorry, I'm on macOS & _not_ a Python developer; I seem to always fight with some versioning issues, unfortunately. Will try again later but if you spot anything wrong in the install output above, I'd be thankful for any hints. The key lines seem to be:

```
ERROR: ERROR: Could not find a version that matches mkdocs-material-extensions>=1.0b1
Skipped pre-versions: 1.0b1, 1.0b1, 1.0b2, 1.0b2
There are incompatible versions in the resolved dependencies.
```

**Comment by squidfunk** [comment]
Before we proceed here, I would recommend trying to deploy on GitHub pages. We need to make sure it's not related to Vercel.

**Comment by borekb** [comment]
If you mean [this](https://www.mkdocs.org/user-guide/deploying-your-docs/), then it requires that I have `mkdocs` locally installed which is currently failing for me (see above). I'll try again later tonight.

**Comment by squidfunk** [comment]
You might try the [official Docker image](https://hub.docker.com/r/squidfunk/mkdocs-material/) if you don't manage to install it in your local environment.

**Comment by squidfunk** [comment]
Otherwise, you seem to be running into the problem that pipenv doesn't install prereleases:
https://pipenv-fork.readthedocs.io/en/latest/diagnose.html

> Pipenv does not install prereleases (i.e. a version with an alpha/beta/etc. suffix, such as 1.0b1) by default. You will need to pass the --pre flag in your command, or set
>
> ```
> [pipenv]
> allow_prereleases = true
> ```

**Comment by borekb** [comment]
> You might try the [official Docker image](https://hub.docker.com/r/squidfunk/mkdocs-material/) if you don't manage to install it in your local environment.

Thanks, I already use Docker for local development, and the issue is not reproducible there as noted above.

Will try the local installation with `--pre` and report back.

**Comment by squidfunk** [comment]
Ah, sorry, I've not seen this comment:

> Since the demo repo deploys fine, can it be that our contents is causing this? Maybe the built-in search indexes some strange unicode character (emoji) and then crashes? But perhaps unlikely because:

This means it's certainly related to your changes and/or some side effect of your changes and instant loading. You could enable instant loading and then add your changes one-by-one to isolate the problem.

**Comment by squidfunk** [comment]
... however, it's still strange that it runs locally then.

**Comment by borekb** [comment]
Ok, I've got a working local setup and I can reproduce the issue on my computer (Vercel not required!).

To sum up:

- `docker run --rm -it -v ${PWD}:/docs squidfunk/mkdocs-material build -d public` builds a folder that I can `cd` into and run via local web server and everything works fine. ✅ 
- `pipenv run mkdocs build -d public` builds a folder that exhibits the strange 100% CPU behavior as reported above 🔴

Some more info:

```console
$ docker run --rm -it -v ${PWD}:/docs squidfunk/mkdocs-material --version
mkdocs, version 1.1 from /usr/local/lib/python3.8/site-packages/mkdocs (Python 3.8)
```

```console
$ pipenv run mkdocs --version
mkdocs, version 1.1.1 from /Users/borekb/.local/share/virtualenvs/docs-AogWyTTU/lib/python3.8/site-packages/mkdocs (Python 3.8)
```

**Comment by squidfunk** [comment]
Could you downgrade your local pip to MkDocs 1.1 and see if the issue persists?

**Comment by borekb** [comment]
I've done `pipenv install mkdocs==1.1` and can still reproduce the issue. My complete versions at this point:

```
$ pipenv run pip freeze
click==7.1.2
future==0.18.2
Jinja2==3.0.0a1
joblib==0.14.1
livereload==2.6.1
lunr==0.5.6
Markdown==3.2.2
MarkupSafe==2.0.0a1
mkdocs==1.1
mkdocs-material==5.1.6
mkdocs-material-extensions==1.0b2
nltk==3.5
Pygments==2.6.1
pymdown-extensions==7.1
PyYAML==5.3.1
regex==2020.5.13
six==1.14.0
tornado==6.0.4
tqdm==4.46.0
```

Would it be useful for you if I shared the entire sources, incl. Markdown files? Or is there anything else I can try, before carefully going through content page by page?

**Comment by borekb** [comment]
Is there a command I could run in the Docker container to see which versions it uses? That is not managed by pip / pipenv, is it?

**Comment by squidfunk** [comment]
The official Docker image hasn't been rebuilt since the latest MkDocs release, so it's definitely using version 1.1. If the issue persists after the local downgrade (as you said), the best thing would be a repository with a minimal use case, which reproduces the error, so I can check it out and debug it, when I find some time. Ideally, you could try stripping stuff step-by-step to isolate the problem, as this would save me a lot of time 😊

**Comment by borekb** [comment]
👍, I'll set myself a reminder to look at it next week (don't have more time right now, unfortunately).

BTW, besides content, I'm also worried that there are so vastly different versions installed in the Docker image and locally (local = left, Docker = right, generated by `docker run --rm -it -v ${PWD}:/docs --entrypoint pip squidfunk/mkdocs-material freeze`):

<img width="1060" alt="Screenshot 2020-05-13 at 22 23 40" src="https://user-images.githubusercontent.com/101152/81861366-6a40f200-9568-11ea-843f-38e4915616ec.png">

Some version differences are semver-major, e.g., Jinja2 or MarkupSafe, which feels strange.

I'd like to install without `--pre` but that's failing for me as noted above.

**Comment by borekb** [comment]
Another observation before I dig deeper:

- ✅ the site works correctly when built with Docker image `squidfunk/mkdocs-material:5.1.5`
- 🚫 the site is broken when built with Docker image `squidfunk/mkdocs-material:5.1.6`

Outputs of these two commands:

```bash
docker run --rm -it -v ${PWD}:/docs --entrypoint pip squidfunk/mkdocs-material:5.1.5 freeze
docker run --rm -it -v ${PWD}:/docs --entrypoint pip squidfunk/mkdocs-material:5.1.6 freeze
```

<img width="1123" alt="Screenshot 2020-05-14 at 12 41 07" src="https://user-images.githubusercontent.com/101152/81925341-a9ab2500-95e0-11ea-88ad-4d8c4cfba87c.png">

**Comment by squidfunk** [comment]
It looks like #1696 is the reason for the observed behavior.

**Comment by borekb** [comment]
Awesome!

**Comment by squidfunk** [comment]
Just released 5.1.7 which fixes #1696 - can you check whether your issue still persists?

**Comment by borekb** [comment]
🎉 both of these work:

- Docker:
  - `docker run --rm -v ${PWD}:/docs squidfunk/mkdocs-material:5.1.7 build`
- Local (not fixed to any particular version but _currently_ works):
  1. `pipenv install mkdocs-material`
  2. `pipenv run mkdocs build`

Thank you!
